CMakeLists.txt

```diff
@@ -95,6 +95,9 @@ if (MSVC)
 		# and https://devblogs.microsoft.com/cppblog/the-fpcontract-flag-and-changes-to-fp-modes-in-vs2022/
 		# By default, MSVC doesn't enable the /fp:fast option.
 		set_cxx_flag("/fp:fast")
+	else()
+		# Precise model (/fp:precise) should do safe contractions, but we should not trust that (see below).
+		set_cxx_flag("/fp:strict")
```

Comment:
`/fp:precise` would be a better default for MSVC. The documentation says `/fp:strict` is mostly needed if you want floating point exceptions. I did a little test with 57 PNG files and got `246.162s` (self-reported) with master and `259.846s` with this branch.